c-api/compat-5.3.c

```diff
@@ -15,6 +15,25 @@
 /* definitions for Lua 5.1 only */
 #if defined(LUA_VERSION_NUM) && LUA_VERSION_NUM == 501
 
+#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (!defined (__MINGW32__) && defined(__GNUC__) && (__GNUC__ < 6))
+#ifndef COMPAT53_HAVE_STRERROR_R
+#define COMPAT53_HAVE_STRERROR_R
+#endif // have strerror_r
+#if ((defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L) || (defined(_XOPEN_SOURCE) || _XOPEN_SOURCE >= 600)) && (!defined(_GNU_SOURCE) || !_GNU_SOURCE)
+#ifndef COMPAT53_HAVE_STRERROR_R_XSI
+#define COMPAT53_HAVE_STRERROR_R_XSI
+#endif // XSI-Compliant strerror_r
+#else
+#ifndef COMPAT53_HAVE_STRERROR_R_GNU
+#define COMPAT53_HAVE_STRERROR_R_GNU
+#endif // GNU variant strerror_r
+#endif // XSI/Posix vs. GNU strerror_r
+#elif defined(_MSC_VER) || (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L) || defined(__STDC_LIB_EXT1__)
```

Comment:
Why `elif`?

Reply:
It either has `strerror_r` or has `strerror_s`: we don't need to know about both.